### PR TITLE
Volunteer check in

### PR DESCRIPTION
**Pull request by zaviermiller**
Added the ability to toggle volunteers attendance, and I made the cards actually link to where they should.

### CODE REVIEW

**Review by andlrutt — APPROVED** [review]
looks good to me!

File: src/pages/admin/events/[eventId]/index.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        },
+    });
+});
+
+export async function getServerSideProps(context: NextPageContext) {
+    // get eventId from url by using context
+    const eventId = context.query.eventId as string;
+
+    // this func is run on server-side, so we can safely fetch the event directly
+    const event: Event = await getEventWithVolFields(eventId);
```
Comment (kemalfidan) [inline src/pages/admin/events/[eventId]/index.tsx]: This is what I was talking about during standup. I think you can use `getEventVolunteers` here. It has pagination and search as well.

**Review by kemalfidan — CHANGES_REQUESTED** [review]
Wrote a comment about what we talked about during standup. I think using `getEventVolunteers` is what we need here.

**Review by kemalfidan — COMMENTED** [review]
I noticed that the page is pretty far different from the Figma design. Specifically, the jumbotron and the search is missing.

Is this PR just for the checkin component? If so just lmk and I can work on the rest.

**Comment by zaviermiller** [comment]
spent some time fixing the bugs @kemalfidan and i discussed last wednesday. Im pretty sure its all working as expected now! there is a lil weirdness with the infinite scroll, but it should be fixable with a `hasMore` field returned with `PaginatedVols` happy to do this!

**Comment by kemalfidan** [comment]
> spent some time fixing the bugs @kemalfidan and i discussed last wednesday. Im pretty sure its all working as expected now! there is a lil weirdness with the infinite scroll, but it should be fixable with a `hasMore` field returned with `PaginatedVols` happy to do this!

Nice solution - I worked on that type so lmk if you want me to work on it!

File: src/pages/admin/events/[eventId]/index.tsx [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+import VolQuickAddDialog from "src/components/VolQuickAddDialog";
+import InfiniteScroll from "react-infinite-scroll-component";
+import constants from "utils/constants";
+
+interface Props {
+    event: Event;
+    pageVols: PaginatedVolunteers;
+}
+
+function orderVolunteers(vols: Volunteer[]): Volunteer[] {
```
Comment (kemalfidan) [inline src/pages/admin/events/[eventId]/index.tsx:42]: Don't think this is used - do you think you can remove this?

File: src/pages/admin/events/[eventId]/index.tsx [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+
+    const createAndRegisterVol = async function (vol: Volunteer) {
+        try {
+            const response = await fetch(`${urls.baseUrl}${urls.api.eventQuickadd(event._id!)}`, {
+                method: "POST",
+                body: JSON.stringify({
+                    ...vol,
+                }),
+            });
+            const responseJSON = (await response.json()) as { success: boolean; payload: unknown };
```
Comment (kemalfidan) [inline src/pages/admin/events/[eventId]/index.tsx:110]: You can use the `APIResponse` type here. I forget we have that pretty often

**Review by kemalfidan — APPROVED** [review]
Looks great! Changes to the menu dropdown are out so that should be good to go too :)

File: src/pages/admin/events/[eventId]/index.tsx (hunk 2 above)
Comment (zaviermiller) [reply src/pages/admin/events/[eventId]/index.tsx:42]: yea def can be removed

File: src/pages/admin/events/[eventId]/index.tsx (hunk 3 above)
Comment (zaviermiller) [reply src/pages/admin/events/[eventId]/index.tsx:110]: :+1:  will commit that to dev

**Pull request merged**